### PR TITLE
Added tutorial notebooks for Focal and Contrastive losses

### PR DESCRIPTION
**Pull request by ashutosh1919**
Related to #1612 .  
@shun-lin - Please review.

### CODE REVIEW

File: docs/tutorials/losses_contrastive.ipynb [hunk 1]
```diff
@@ -0,0 +1,236 @@
+{
```
Comment (shun-lin) [inline docs/tutorials/losses_contrastive.ipynb:1]: <p>This block is no longer needed, colab is using 2.x by default :)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/addons/pull/1732/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

File: docs/tutorials/losses_focal.ipynb [hunk 2]
```diff
@@ -0,0 +1,236 @@
+{
```
Comment (shun-lin) [inline docs/tutorials/losses_focal.ipynb:1]: <p>Same, this block is unnecessary :) </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/addons/pull/1732/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

**Comment by ashutosh1919** [comment]
@shun-lin , On loading the vecs.tsv in the UMAP of the projector, it is showing parsing failed. Please can you help me?

**Comment by ashutosh1919** [comment]
focal_loss_image
Focal loss image

**Comment by ashutosh1919** [comment]
contrastive_loss
Contrastive loss image

**Comment by ashutosh1919** [comment]
@shun-lin, I am not getting why the tests are failing after merging with master. Please take a look at it. I have added images for embeddings.

**Comment by shun-lin** [comment]
@ashutosh1919 sorry I didn't see your comment earlier as I was working, but would you be able to download the embedding and upload it to tensorboard public server (similar to Triplet Loss's tutorial)? Also the embedding images look nice! would you able to color code it? (without color coding it's harder to see whether or not it works haha).

I am also aware that there are also another PR that is working on tutorial of focal loss, it that also your PR?

**Comment by shun-lin** [comment]
> @shun-lin, I am not getting why the tests are failing after merging with master. Please take a look at it. I have added images for embeddings.

Hmm I am not sure, I will tag @seanpmorgan :)

**Comment by ashutosh1919** [comment]
> I am also aware that there are also another PR that is working on tutorial of focal loss, it that also your PR?
No, I haven't made any separate PR for focal_loss. Please, mention that PR if possible.

**Comment by seanpmorgan** [comment]
@googlebot I consent.

**Comment by shun-lin** [comment]
@ashutosh1919 would you able to create embedding with colors for different classes?

**Comment by ashutosh1919** [comment]
> @ashutosh1919 would you able to create embedding with colors for different classes?

@shun-lin , Not really, I am not even able to load the vectors in the projector. Everytime, it is showing parsing failed.

**Comment by shun-lin** [comment]
> > @ashutosh1919 would you able to create embedding with colors for different classes?
> 
> @shun-lin , Not really, I am not even able to load the vectors in the projector. Everytime, it is showing parsing failed.

interesting, bugs at TensorBoard's side?

**Comment by ashutosh1919** [comment]
@shun-lin , I am not quite sure about that. Can you please try once if you have sometime.

**Comment by ashutosh1919** [comment]
Screenshot 2021-05-13 at 2 03 13 PM
Contrastive Loss

**Comment by ashutosh1919** [comment]
Screenshot 2021-05-13 at 2 38 14 PM
Focal Loss

**Comment by ashutosh1919** [comment]
@shun-lin , @seanpmorgan , @bhack , @gabrieldemarmiesse - I have added embedding image where all labels have different color. Please review.

**Comment by bhack** [comment]
Now we have also a contrastive keras/Addons tutorial at https://keras.io/examples/vision/supervised-contrastive-learning/

We could think to expand that one?

**Comment by ashutosh1919** [comment]
IMO, we should have different tutorial for contrastive learning. This PR contains information about contrastive loss and other code (Model building, compiling and running) is easy to understand. If we expand this tutorial to have Supervised contrastive learning, then it may be difficult for beginners to understand. 
@bhack , I would appreciate your opinion on this.

**Comment by bhack** [comment]
We are collecting more tutorial in the Keras.io repo so I suggest you to port this PR to https://github.com/keras-team/keras-io
